zerver/lib/bugdown/__init__.py

```diff
@@ -1418,6 +1452,9 @@ def bugdown_stats_finish():
 
 def convert(content, message=None, message_realm=None, possible_words=None, sent_by_bot=False):
     # type: (Text, Optional[Message], Optional[Realm], Optional[Set[Text]], Optional[bool]) -> Text
+    global current_message_realm_id
+    if message_realm is not None:
+        current_message_realm_id = message_realm.id
```

Comment:
(Also, it's important that database queries are not done inside bugdown, which is why the function should pass in the Realm object that was in turn passed into Bugdown (if any; if no Realm is passed into bugdown, that's a test suite thing, and we should just follow the server default setting))